ui/components/MesheryApplications.js

```diff
@@ -248,21 +249,53 @@ function MesheryApplications({
     /**  @type {TypeView} */
     ("grid")
   );
-
+  const disposeConfSubscriptionRef = useRef(null);
   const searchTimeout = useRef(null);
 
   /**
    * fetch applications when the page loads
    */
   useEffect(() => {
-    fetchApplications(page, pageSize, search, sortOrder);
   }, [page, pageSize, search, sortOrder]);
 
   /**
    * fetch applications when the application downloads
    */
   useEffect(() => {
+    const configurationSubscription = ConfigurationSubscription((result) => {
+      setPage(result.configuration?.applications.page || 0);
+      setPageSize(result.configuration?.applications.page_size || 0);
+      setCount(result.configuration?.applications.total_count || 0);
+      setApplications(result.configuration?.applications.applications)
+    },
+    {
+      applicationSelector : {
+        pageSize : pageSize.toString(),
+        page : page.toString(),
+        search : search,
+        order : sortOrder
+      },
+      patternSelector : {
+        pageSize : pageSize.toString(),
+        page : page.toString(),
+        search : search,
+        order : sortOrder
+      },
+      filterSelector : {
+        pageSize : pageSize.toString(),
+        page : page.toString(),
+        search : search,
+        order : sortOrder
+      }
+    });
```

Comment:
@MUzairS15 I guess you wrote the same function here like the initAppsSubscription (I might be wrong)

Reply:
Oops, you are right

Reply:
Glad that i can help :handshake: